### PR TITLE
[dag] remove support for weak links for now

### PR DESCRIPTION
**Pull request by zekun000**

### CODE REVIEW

File: consensus/src/dag/dag_store.rs [hunk 1]
```diff
@@ -226,46 +215,23 @@ impl Dag {
...
-                            Some(&node.metadata.author)
-                        } else {
-                            None
-                        }
-                    })
-                });
+        let all_nodes_in_round = self.nodes_by_round.get(&round)?.iter().flatten();
         if validator_verifier
-            .check_voting_power(strong_link_authors)
+            .check_voting_power(all_nodes_in_round.clone().map(|node| &node.metadata.author))
```
Comment (ibalajiarun) [inline consensus/src/dag/dag_store.rs:225]: nit: can we clone the vec<author> instead? Not sure which is better.

File: consensus/src/dag/dag_store.rs [hunk 2]
```diff
@@ -226,46 +215,23 @@ impl Dag {
         self.nodes_by_digest.get(digest).cloned()
     }
 
-    pub fn get_unlinked_nodes_for_new_round(
+    pub fn get_strong_links_for_round(
         &self,
+        round: Round,
         validator_verifier: &ValidatorVerifier,
     ) -> Option<Vec<NodeMetadata>> {
```
Comment (ibalajiarun) [inline consensus/src/dag/dag_store.rs:222]: no strong opinion but I feel like Vec<NodeMetadata> should be a type called RoundParents.

**Review by ibalajiarun — APPROVED** [review]

File: consensus/src/dag/dag_store.rs (hunk 1 above)
Comment (zekun000) [reply consensus/src/dag/dag_store.rs:225]: it's a flat map type, not the vector, I was trying to save the iter().flattern() piece

**Review by sasha8 — APPROVED** [review]
LGTM

**Pull request closed without merging**